src/twisted/test/test_defer.py

```diff
@@ -1823,6 +1855,66 @@ def test_canceller_circular_reference_non_final(self) -> None:
         # been freed.
         self.assertIsNone(weakCanceller())
 
+    @pyunit.skipIf(_PYPY, "GC works differently on PyPy.")
+    def test_inlineCallbacksNoCircularReference(self) -> None:
+        """
+        Tests that there is no circular dependency when using
+        L{defer.inlineCallbacks}, so that the machinery gets cleaned up
+        immediately rather than waiting for a GC.
+        """
```

Comment:
Maybe something like this, focusing on the behaviour that is tested here.

We don't have any code in the test that looks into the GC part.

```suggestion
        """
        When using L{defer.inlineCallbacks},
        after the function exits,
        it will not keep references to the function itself or the arguments.
        """
```